server/src/main/java/io/spine/server/stand/SubscriptionMatcher.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2019, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.stand;
+
+import com.google.protobuf.Any;
+import com.google.protobuf.Message;
+import io.spine.base.FieldPath;
+import io.spine.client.CompositeFilter;
+import io.spine.client.CompositeFilter.CompositeOperator;
+import io.spine.client.Filter;
+import io.spine.client.IdFilter;
+import io.spine.client.Subscription;
+import io.spine.client.Target;
+import io.spine.client.TargetFilters;
+import io.spine.core.EventEnvelope;
+import io.spine.protobuf.TypeConverter;
+import io.spine.type.TypeUrl;
+
+import java.util.function.Predicate;
+
+import static io.spine.protobuf.FieldPaths.fieldAt;
+import static io.spine.server.storage.OperatorEvaluator.eval;
+import static io.spine.util.Exceptions.newIllegalArgumentException;
+import static java.lang.String.join;
+
+/**
+ * Matches the incoming event against a subscription criteria.
+ *
+ * <p>The class is abstract because event and entity subscriptions match through the different
```

Comment:
To me this paragraph should say that descendants decide on how to turn the incoming event envelope into a set of data pieces suitable for filtering.

And not that "this class is abstract because etc".